src/averageAges.js

```diff
@@ -1,61 +1,63 @@
 'use strict';
 
 /**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menInCentury = men.filter(man => Math.ceil(man.died / 100) === century);
+  const menCalculated = century ? menInCentury : men;
+  const menAge = menCalculated.map(man => man.died - man.born);
+
+  return menAge.reduce((sum, n) => sum + n) / menAge.length;
```

Comment:
let's create separate function that will accept people array and will return average age value